edi_voxel_sale_order_import_oca/models/sale_order.py

```diff
@@ -0,0 +1,384 @@
+                error_msgs.append(
+                    _(
+                        "Couldn't find any <b>Delivery Address</b> "
+                        "corresponding to the following information extracted "
+                        "from the Voxel document:<br/>"
+                        "<ul>%s</ul>"
+                    )
+                    % (msg_fields)
+                )
+
+    def _parse_partner_data_voxel(self, data, raise_error=True):
+        domains = []
+        # fill domains list
+        if data.get("ref"):
+            domains.append([("ref", "=", data.get("ref"))])
+        partner = self.env["res.partner"].search(expression.AND(domains))
+        if len(partner) == 1:
+            return partner
+        elif len(partner) == 0:
+            domains = []
+        # If there are ZERO or more than ONE objects, continue filling domains
+        if data.get("country_id"):
+            domains.append([("country_id.code_alpha3", "=", data["country_id"])])
+        if data.get("state_id"):
+            domains.append([("state_id.name", "=", data["state_id"])])
+        if data.get("name"):
+            domains.append([("name", "=", data["name"])])
+        if data.get("email"):
+            domains.append([("email", "=", data["email"])])
+        if data.get("zip"):
+            domains.append([("zip", "=", data["zip"])])
+        elif data.get("city"):
+            domains.append([("city", "=", data["city"])])
+        partner = self.env["res.partner"].search(expression.AND(domains))
+        if len(partner) == 1:
+            return partner  # return the unique partner matching
+        if raise_error:
+            raise UserError(
+                _(
+                    "Can't find a suitable partner for this data:\n\n%(data)s"
+                    "\nResults: %(partner_count)s",
+                    data=data,
+                    partner_count=len(partner),
+                )
+            )
+        return self.env["res.partner"]
+
+    def _parse_comments_data_voxel(self, vals, xml_root, error_msgs):
+        comments = []
+        comment_elements = xml_root.xpath("//Comments/Comment")
+        for comment_element in comment_elements:
+            comment_data = comment_element.attrib
+            subject = comment_data.get("Subject")
+            msg = comment_data.get("Msg")
+            comments.append(":\n".join(filter(None, [subject, msg])))
+        if comments:
+            vals.update(note="\n\n".join(comments))
+
+    def _parse_product_list_data_voxel(self, order, xml_root, error_msgs):
+        line_elements = xml_root.xpath("//ProductList/Product")
+        so_line_obj = self.env["sale.order.line"]
+        for line_element in line_elements:
+            line_vals = {"order_id": order.id}
+            self._parse_product_voxel(line_vals, line_element)
+            self._parse_qty_uom_voxel(line_vals, line_element)
+            line_vals = so_line_obj.play_onchanges(line_vals, list(line_vals)[1:])
```

Comment:
Include the first value (order_id) to take into account related values like company_id when applying changes.

```suggestion
            line_vals = so_line_obj.play_onchanges(line_vals, list(line_vals))
```

Reply:
Changes done!!